message_sender/views.py

```diff
@@ -137,13 +138,17 @@ def post(self, request, *args, **kwargs):
                                 request.data["nack_reason"]
                             message.save()
                         send_message.delay(str(message.id))
-
                         if "voice_speech_url" in message.metadata:
                             fire_metric.apply_async(kwargs={
                                 "metric_name":
                                     'vumimessage.obd.unsuccessful.sum',
                                 "metric_value": 1.0
                             })
+                    outbound_type = "voice" if "voice_speech_url" in \
+                        message.metadata else "text"
+                    ConcurrencyLimiter.decr_message_count(
+                        outbound_type, message.last_sent_time)
```

Comment:
This is for an acknowledge, which happens immediately after we connect the call, not when the call is complete. I think what we want is to rather look at the inbound messages, and look for an inbound message with a session end event.

If you look at the freeswitch transport as an example, we send an ack/nack as soon as we start the playback, but send an inbound session close message once we receive a hangup event: https://github.com/praekelt/vumi-freeswitch-esl/blob/develop/vxfreeswitch/voice.py#L399-L400